protostar/commands/test/cheatcodes/reflect_cheatcode.py

```diff
@@ -0,0 +1,224 @@
+
+            assert isinstance(curr._struct_definition, StructDefinition)
+            assert isinstance(curr._reference_value, RelocatableValue)
+
+            name: str
+            member_definition: MemberDefinition
+            for name, member_definition in curr._struct_definition.members.items():
+                expr_type: CairoType = member_definition.cairo_type
+                addr = curr._reference_value + member_definition.offset
+
+                if is_simple_type(expr_type):
+                    if isinstance(expr_type, TypeFelt):
+                        tmp = int(curr.get_or_set_value(name, None))  # type: ignore
+                        assert isinstance(tmp, int)
+                        curr_dict[name] = tmp
+                    else:
+                        assert isinstance(expr_type, TypePointer)
+                        tmp = deepcopy(curr.get_or_set_value(name, None))
+                        assert isinstance(tmp, RelocatableValue)
+                        curr_dict[name] = tmp
+
+                elif isinstance(expr_type, TypeStruct):
+                    curr_dict[name] = _ReflectTreeNode(
+                        typename=expr_type.scope.path[1], value={}
+                    )
+                    stack.append(
+                        (
+                            curr_dict[name].value,
+                            VmConstsReference(
+                                context=curr._context,
+                                struct_name=expr_type.scope,
+                                reference_value=addr,
+                            ),
+                        )
+                    )
+
+                else:
+                    assert isinstance(expr_type, TypePointer) and isinstance(
+                        expr_type.pointee, TypeStruct
+                    ), "Type must be of the form T*."
+
+                    tmp = deepcopy(curr._context.memory[addr])  # type: ignore
+                    assert isinstance(tmp, RelocatableValue)
+                    curr_dict[name] = tmp
+
+        assert isinstance(value._struct_definition, StructDefinition)
+        return _ReflectTreeNode(
+            typename=value._struct_definition.full_name.path[1], value=value_tree
+        )
+
+    # pylint: disable=R0201
+    def _convert_to_named_tuple(self, tree: _ReflectTreeNode) -> REFLECT_RETURN_TYPE:
+        stack: List[REFLECT_VALUE_TYPE] = [tree]
+        pre_order: List[REFLECT_VALUE_TYPE] = []
+
+        while len(stack) > 0:
+            curr = stack.pop()
+            pre_order.append(curr)
+            if isinstance(curr, _ReflectTreeNode):
+                assert isinstance(curr.value, dict)
+                for _, node in curr.value.items():
+                    stack.append(node)
+
+        # When we reach each element it contains a single value or namedtuple instead of a dict
+        for node in reversed(pre_order):
+            if not isinstance(node, _ReflectTreeNode):
```

Comment:
Can we prefilter those instead of this conditional?
`[x for x if not isinstance]`